Fix for test script ScriptCanvasComponent_OnEntityActivatedDeactivated_PrintMessage

michleza opened the pull request:
Signed-off-by: Michael <michleza@amazon.com>

## What does this PR do?
Migrated the test to hydra paradigm
Updated test to use scripting tools file and constants
Reformatted confusing syntax to make test more readable
Removed unnecessary code
Fixed outdated component property paths on entity script canvas components
Fixed broken script canvas files used by test so test passes when game mode is entered & exited
Reactivated the test in periodic test suite

FuzzyCarterAWS commented on AutomatedTesting/Gem/PythonTests/scripting/ScriptCanvasComponent_OnEntityActivatedDeactivated_PrintMessage.py:
```diff
@@ -134,55 +130,51 @@ def validate_start_state(entity: EditorEntity, expected_state: str):
             entity.set_start_status(expected_state)
         return True
```
This will always return true as written which means that any of the logic used from this function will always be true if it exits. Otherwise we're raising errors. This function likely needs to be re-thought out.

FuzzyCarterAWS replied:
Additionally, this function should be a little more meaningful `validate_entity_state` or `validate_entity_start_state`.

Lastly, is this function also another scripting_tools function candidate?

michleza replied:
I migrated this to the tools file and updated the return statement to check if the expected state is equal to the actual state. We  need to have a quick chat about the 'state_options' variable though. Originally, it was drawing from a global variables file. I spent some time trying to get the global variables to work with the script tools file but it just wouldn't. I'm not sure if the solution I have here is safe,

FuzzyCarterAWS commented on AutomatedTesting/Gem/PythonTests/scripting/ScriptCanvasComponent_OnEntityActivatedDeactivated_PrintMessage.py:
```diff
@@ -45,77 +72,46 @@ def ScriptCanvasComponent_OnEntityActivatedDeactivated_PrintMessage():
+        self.create_editor_entity(deactivated_dict)
+        self.create_editor_entity(controller_dict)
+
+    def create_editor_entity(self, entity_dict: dict):
+        """
+        Create entities and configure their activation status and variables. We call two different libraries to modify the
+        entity. Hydra editor utils to create the entity and editor entity utils to drill into its properties to change values.
+
+        """
+        entity = scripting_tools.create_entity_with_sc_component_asset(entity_dict["name"], entity_dict["path"])
+        editor_entity = EditorEntity.find_editor_entity(entity_dict["name"])
+        editor_entity.set_start_status(entity_dict["status"])
+
+        if entity_dict["name"] == "Controller":
+            activated_entity = EditorEntity.find_editor_entity(activated_dict["name"])
+            sc_component = editor_entity.get_components_of_type(["Script Canvas"])[0]
+            sc_component.set_component_property_value(ENTITY_TO_ACTIVATE_PATH, activated_entity.id)
+
+            deactivated_entity = EditorEntity.find_editor_entity(deactivated_dict["name"])
+            sc_component = editor_entity.get_components_of_type(["Script Canvas"])[0]
+            sc_component.set_component_property_value(ENTITY_TO_DEACTIVATE_PATH, deactivated_entity.id)
+
+        helper.wait_for_condition(lambda: entity is not None, WAIT_TIME_3)
```
Scripting Tools function candidate?

michleza replied:
migrated to the tools file

FuzzyCarterAWS commented on AutomatedTesting/Gem/PythonTests/scripting/ScriptCanvasComponent_OnEntityActivatedDeactivated_PrintMessage.py:
```diff
@@ -45,77 +72,46 @@ def ScriptCanvasComponent_OnEntityActivatedDeactivated_PrintMessage():
+        helper.wait_for_condition(lambda: entity is not None, WAIT_TIME_3)
+
+    def validate_entity_exist(self, entity_name: str, test_tuple: tuple):
         """
         Validate the entity with the given name exists in the level
         :return: entity: editor entity object
         """
-        entity = Entity.find_editor_entity(entity_name)
+        entity = EditorEntity.find_editor_entity(entity_name)
         Report.critical_result(test_tuple, entity.id.IsValid())
         return entity
```
Scripting tools function canadidate?

michleza replied:
migrated to the tools file

FuzzyCarterAWS commented on AutomatedTesting/Gem/PythonTests/scripting/ScriptCanvasComponent_OnEntityActivatedDeactivated_PrintMessage.py:
```diff
@@ -134,55 +130,51 @@ def validate_start_state(entity: EditorEntity, expected_state: str):
+        controller = self.validate_entity_exist(controller_dict["name"], Tests.controller_exists)
+        state1_correct = self.validate_start_state(controller, controller_dict["status"])
 
-        act_tester = validate_entity_exist(activated_dict["name"], Tests.activated_exists)
-        state2_correct = validate_start_state(act_tester, activated_dict["status"])
+        act_tester = self.validate_entity_exist(activated_dict["name"], Tests.activated_exists)
+        state2_correct = self.validate_start_state(act_tester, activated_dict["status"])
 
-        deac_tester = validate_entity_exist(deactivated_dict["name"], Tests.deactivated_exists)
-        state3_correct = validate_start_state(deac_tester, deactivated_dict["status"])
+        deac_tester = self.validate_entity_exist(deactivated_dict["name"], Tests.deactivated_exists)
+        state3_correct = self.validate_start_state(deac_tester, deactivated_dict["status"])
 
         all_states_correct = state1_correct and state2_correct and state3_correct
         Report.critical_result(Tests.start_states_correct, all_states_correct)
```
Scripting tools function candidate?

michleza replied:
migrated this function to the tools file

michleza commented:
updated based on Fuzzy's feedback. I want to discuss the state_options variable since originally it was reaching out to a global variables class but that class seems inaccessible outside of that very specific context it was being used. I need help understanding why I can't call out to it and get the properties from the scripting tools file.

michleza commented:
updated again based on discussions with Fuzzy. Renamed a test script function and removed it's logic to make it more understandable. Entity setup is much more straightforward now. We were unable to figure out why the globals file properties are accessible within their original context but not anywhere else, even if you replace the function they are used with another function that uses them.

FuzzyCarterAWS commented on AutomatedTesting/Gem/PythonTests/scripting/ScriptCanvasComponent_OnEntityActivatedDeactivated_PrintMessage.py:
```diff
@@ -45,144 +72,77 @@ def ScriptCanvasComponent_OnEntityActivatedDeactivated_PrintMessage():
+
+    def validate_entities_in_level(self):
+
+        controller = scripting_tools.validate_entity_exists_by_name(controller_dict["name"], Tests.controller_exists)
+        state1_correct = scripting_tools.validate_entity_start_state_by_name(controller, controller_dict["status"])
```
This code repeats for each entity you're trying to validate.

I'd recommend extracting this behavior into another function then just pass in the values you're trying to validate.

This will reduce the number of lines of code, remove repeated code, and make it clearer what you're trying to do.

michleza replied:
broke this down into two new functions and made the function calling the tools file a bit more generic
